opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/service/consent/authentication/HbciInitiateSendPinAndPsuId.java

```diff
@@ -20,39 +22,62 @@
 
 @Service("hbciInitiateSendPinAndPsuId")
 @RequiredArgsConstructor
+@Slf4j
 public class HbciInitiateSendPinAndPsuId extends ValidatedExecution<HbciContext> {
 
     private final Optional<HBCIProduct> product;
     private final OnlineBankingService onlineBankingService;
+    private final HbciAuthorizationPossibleErrorHandler errorSink;
+
 
     @Override
     protected void doRealExecution(DelegateExecution execution, HbciContext context) {
-        HbciConsent consent = context.getHbciDialogConsent();
-        if (null == consent) {
-            consent = new HbciConsent();
-            consent.setHbciProduct(product.orElse(null));
-            consent.setCredentials(Credentials.builder()
-                    .userId(context.getPsuId())
-                    .pin(context.getPsuPin())
-                    .build()
-            );
-        }
 
-        UpdatePsuAuthenticationRequest request = new UpdatePsuAuthenticationRequest();
-        request.setCredentials(consent.getCredentials());
-        request.setBankApiConsentData(consent);
-        request.setBank(context.getBank());
+        errorSink.handlePossibleAuthorizationError(
+                () -> {
+                    context.setWrongAuthCredentials(false);
```

Comment:
Suggest to extract (Runnable) lambda function body to the private method as it is quite long

Reply:
done